ingest/writer_functions.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import annotations
+from typing import TextIO
+import re
+import os
+import gzip
+import json
+
+try:
+    from ingest_files import IngestFiles
+except ImportError:
+    from .ingest_files import IngestFiles
+
+# regex to split on commas and tabs
+COMMA_OR_TAB = r"[,\t]"
+
+# regex to split on comma/tab/space
+ALL_DELIM = r"[\s,\t]"
+
+# Gzip magic number
+GZIP_MAGIC_NUM = b'\x1f\x8b'
+
+ALLOWED_FILE_TYPES = ["text/csv", "text/plain", "text/tab-separated-values"]
+
+def is_gz_file(filepath) -> bool:
+    """
+    Determine if a file is gzipped by checking first two bytes against GZIP_MAGIC_NUM
+
+    :param filepath: (str) path to file to check
+    :returns: (bool)
+    """
+    with open(filepath, 'rb') as test_f:
+        return test_f.read(2) == GZIP_MAGIC_NUM
+
+def round_exp(val, precision) -> float:
+    """
+    Round a raw value to a specified precision
+
+    :param val: (str) raw value to convert and round
+    :param precision: (int) number of digits of precision
+    :returns: (float)
+    """
+    return round(float(val), precision)
+
+def encode_cluster_name(name) -> str:
+    """
+    Encodes a cluster name to be used in a HTTP request
+    Will encode + signs as 'pos' to differentiate + and -
+    :param name: (str) name of cluster
+    :returns: (str) encoded cluster name
```

Comment:
Having the return type redundantly noted in a docstring means it could get out-of-sync -- a mild code smell.

```suggestion
    :returns: encoded cluster name
```